src/aliceVision/sfm/pipeline/sequential/ReconstructionEngine_sequentialSfM.cpp

```diff
@@ -187,96 +190,110 @@ void ReconstructionEngine_sequentialSfM::RobustResectionOfImages(
+        ++imageIndex;
+        
         {
-          // Some views can become indirectly localized when the sub-pose becomes defined
-          if(_sfm_data.IsPoseAndIntrinsicDefined(view.getViewId()))
+          const View& view = *_sfm_data.views.at(possible_resection_index);
+          
+          if(view.isPartOfRig())
           {
-            ALICEVISION_LOG_DEBUG("Resection of image " << currentIndex << " ID=" << possible_resection_index << " was skipped." << std::endl
-                              << "RigID=" << view.getRigId() << " Sub-poseID=" << view.getSubPoseId()
-                              << " sub-pose and pose defined.");
-            set_remainingViewId.erase(possible_resection_index);
-            continue;
+            // Some views can become indirectly localized when the sub-pose becomes defined
+            if(_sfm_data.IsPoseAndIntrinsicDefined(view.getViewId()))
+            {
+              ALICEVISION_LOG_DEBUG("Resection of image " << currentIndex << " ID=" << possible_resection_index << " was skipped." << std::endl
+                                    << "RigID=" << view.getRigId() << " Sub-poseID=" << view.getSubPoseId()
+                                    << " sub-pose and pose defined.");
+#pragma omp critical                              
+              set_remainingViewId.erase(possible_resection_index);
+              
+              continue;
+            }
+            
+            // We cannot localize a view if it is part of an initialized RIG with unknown Rig Pose
+            const bool knownPose = _sfm_data.existsPose(view);
+            const Rig& rig = _sfm_data.getRig(view);
+            const RigSubPose& subpose = rig.getSubPose(view.getSubPoseId());
+            
+            if(rig.isInitialized() &&
+               !knownPose &&
+               (subpose.status == ERigSubPoseStatus::UNINITIALIZED))
+            {
+              ALICEVISION_LOG_DEBUG("Resection of image " << currentIndex << " ID=" << possible_resection_index << " was skipped." << std::endl
+                                    << "RigID=" << view.getRigId() << " Sub-poseID=" << view.getSubPoseId()
+                                    << " Rig initialized but unkown pose and sub-pose.");
+#pragma omp critical   
+              set_remainingViewId.erase(possible_resection_index);
+              
+              continue;
+            }
           }
-
-          // We cannot localize a view if it is part of an initialized RIG with unknown Rig Pose
-          const bool knownPose = _sfm_data.existsPose(view);
-          const Rig& rig = _sfm_data.getRig(view);
-          const RigSubPose& subpose = rig.getSubPose(view.getSubPoseId());
-
-          if(rig.isInitialized() &&
-             !knownPose &&
-             (subpose.status == ERigSubPoseStatus::UNINITIALIZED))
+        }
+        
+        bool bResect = false;
+#pragma omp critical      
+        {
+          bResect = Resection(possible_resection_index);
```

Comment:
The computation of the resectioning in itself is independent for each view.
So we need to split the Resection into 2 methods to get performance improvements.
Something like:
```
ResectionResult newResection;
bool valid = computeResection(view, newResection);
if(valid)
{
#pragma omp critical
  {
    updateScene(newResection);
    nbViewsAdded += 1;
  }
}
```